packages/back-end/src/services/organizations.ts

```diff
@@ -97,20 +99,34 @@ export function validateLoginMethod(
   return true;
 }
 
-export function getOrgFromReq(req: AuthRequest) {
+export type ReqContext = {
+  org: OrganizationInterface;
+  userId: string;
+  email: string;
+  environments: string[];
+  userName: string;
+  readAccessFilter: ReadAccessFilter;
+};
+
+export async function getContextFromReq(req: AuthRequest): Promise<ReqContext> {
   if (!req.organization) {
     throw new Error("Must be part of an organization to make that request");
   }
   if (!req.userId || !req.email) {
     throw new Error("Must be logged in");
   }
 
+  const teams = await getTeamsForOrganization(req.organization.id);
```

Comment:
I looked into how we could keep `getContextFromReq` from being `async` and it looks like we already are loading `teams` in our auth middleware, just not attaching it to the req object.

https://github.com/growthbook/growthbook/blob/0b77e87f936c0a9fd319a4f4f4df6e52bee5dd8e/packages/back-end/src/services/auth/index.ts#L178

What do you think about adding this to `req` via `req.teams` and using that to generate the read access filter? That should allow us to keep this method synchronous

Reply:
You bet - I went back and forth on this. I've added `teams` to the `req` object as requested!